### ISSUE TITLE
TMC2208 is switching to SpreadCycle with Linear Advance active and HYBRID_THRESHOLD disabled

### ISSUE THREAD

**Issue by lienbacher** [teemuatlut/Marlin#20]
Description: I just tried linear advance on my printer using marlin firmware and I noticed weird sound when the printer is actually working near edges, it was also audible when I printed the K-Factor calibration, especially in acceleration and deceleration phases. I started investigating and sending M122 in the right moment actually reveals the driver seems to switch back and forth between SpreadCycle/StealthChop, most of the time StealthChop is returned true via M122, but in some cases, especially during times when there is a lot of Linear Advance magic happening it returns false. I tried to read the code to find if this behavior is intended, but was not successfull in finding any hints. 
Initially I thought this was a behaviour caused by HYBRID_THRESHOLD, but it is disabled, so I am left without clues what could cause this the mode-switching.
I can try taking a video or audio recording of the sound produced by my Y axis tomorrow. Am still using one single TMC2208 along with 2 TMC2010 on X and Z, E is still a 4899. It is quite audible that the weird sound comes from the Y-Stepper. Without linear advance enabled there is no such behavior.

@teemuatlut, maybe you have more clue about what could happen there and why?
thanks!

**Comment by lienbacher** [comment]
Now that I got some sleep I have the impression I might be on a very wrong path. I'll update this when I get to investigate it further. It may indeed be the extruder motor doing the funky noises, which is still on a 4899 ...

**Comment by lienbacher** [comment]
TL:DR, it was the Y-Motors fault. I changed it against a higher quality motor and the sound was much reduced. Even though not completely gone, I am now 100% certain it is not an issue of the TMC code in any way. Sorry for the fuzz!

I did a lot of tests and also changed drivers of different axis around to make sure it is not the driver. Eventually I was certain enough the very choppy almost breaking mechanical noise was coming from  the Y-Motor. I intentionally switched StealthChop off and it did the same breaking noise in SpreadCycle, still sounded a lot different to StealthChop. Also turned HYBRID_THRESHOLD off to rule out it was not the driver switching to Spreadcycle and the sound stayed the same. After changing the motor to a higher quality and higher current stepper the noise was very much reduced, but it pretty much sounds like the X-Axis now which never was a concern.
